pinot-query-planner/src/main/java/org/apache/pinot/calcite/rel/rules/PinotWindowExchangeNodeInsertRule.java

```diff
@@ -234,8 +283,13 @@ private void validateWindowFrames(Window.Group windowGroup) {
     }
   }
 
-  private boolean isRowsOnlyAggregationCallType(ImmutableList<Window.RexWinAggCall> aggCalls) {
-    return aggCalls.stream().anyMatch(aggCall -> aggCall.getKind().equals(SqlKind.ROW_NUMBER));
+  private boolean hasRankBasedAgg(ImmutableList<Window.RexWinAggCall> aggCalls) {
+    for (Window.RexWinAggCall aggCall : aggCalls) {
+      if (RANK_BASED_FUNCTION_KIND.contains(aggCall.getKind())) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
I just discovered that we probably don't need any of the `hasRankBasedAgg` checks at all because Calcite itself ensures that functions like `RANK` / `DENSE_RANK` / `ROW_NUMBER` aren't used with a custom `ROWS` / `RANGE` window frame - https://github.com/apache/calcite/blob/df2328b7c2834cf7ffbf7305fe2c7aec5d07fe64/core/src/main/java/org/apache/calcite/sql/SqlWindow.java#L669-L674.

Reply:
Good call. Let's add a test to ensure it doesn't get planned